tests/eos-test-utils.c

```diff
@@ -0,0 +1,1960 @@
+
+  *out_file = g_file_get_child (eos_dir, rel_path);
+  *out_signature = g_file_get_child (eos_dir, sig_rel_path);
+}
+
+static gboolean
+gpg_sign (GFile *file,
+          GFile *signature,
+          const gchar *keyid,
+          CmdStuff *cmd,
+          GError **error)
+{
+  SSDEF;
+  g_auto(GStrv) argv = NULL;
+
+  argv = generate_strv (GPG_BINARY,
+                        "--homedir=" GPG_HOME_DIRECTORY,
+                        SS (flag ("default-key", keyid)),
+                        SS (flag ("output", SS (g_file_get_path (signature)))),
+                        "--detach-sig",
+                        SS (g_file_get_path (file)),
+                        NULL);
+
+  if (!rm_rf (signature, error))
+    return FALSE;
+
+  return test_spawn (argv, NULL, cmd, error);
+}
+
+static gboolean
+generate_ref_file (GFile *repo,
+                   const gchar *ref,
+                   const gchar *commit,
+                   const gchar *keyid,
+                   GError **error)
+{
+  g_autoptr(GFile) ref_file = NULL;
+  g_autoptr(GFile) ref_file_sig = NULL;
+  g_autoptr(GFile) ref_file_parent = NULL;
+  g_autoptr(GBytes) bytes = NULL;
+  g_auto(CmdStuff) cmd = CMD_STUFF_CLEARED;
+
+  get_ref_file_paths (repo,
+		      ref,
+		      &ref_file,
+		      &ref_file_sig);
+  ref_file_parent = g_file_get_parent (ref_file);
+
+  if (!create_directory (ref_file_parent,
+                         error))
+    return FALSE;
+
+  bytes = g_bytes_new_static (commit, strlen (commit));
+  if (!create_file (ref_file, bytes, error))
+    return FALSE;
+
+  if (!gpg_sign (ref_file,
+                 ref_file_sig,
+                 keyid,
+                 &cmd,
+                 error))
+    return FALSE;
+
+  return cmd_stuff_ensure_ok (&cmd, error);
+}
+
+static gboolean
+update_commits (EosTestSubserver *subserver,
+		GError **error)
+{
+  GHashTableIter iter;
+  gpointer ref_ptr;
+  gpointer commit_ptr;
+  g_auto(CmdStuff) cmd = CMD_STUFF_CLEARED;
+
+  g_hash_table_iter_init (&iter, subserver->ref_to_commit);
+  while (g_hash_table_iter_next (&iter, &ref_ptr, &commit_ptr))
+    {
+      guint commit = GPOINTER_TO_UINT (commit_ptr);
+      g_autofree gchar *checksum = NULL;
+
+      if (!prepare_commit (subserver->repo,
+                           subserver->tree,
+                           commit,
+                           ref_ptr,
+                           subserver->keyid,
+                           &checksum,
+                           error))
+        return FALSE;
+
+      if (!generate_ref_file (subserver->repo,
+                              ref_ptr,
+                              checksum,
+                              subserver->keyid,
+                              error))
+        return FALSE;
+    }
+
+  if (!ostree_summary (subserver->repo,
+                       subserver->keyid,
+                       &cmd,
+                       error))
+    return FALSE;
+  if (!cmd_stuff_ensure_ok (&cmd, error))
```

Comment:
return the value of `cmd_stuff_ensure_ok`

Side note: Is there any way to make the command and `cmd_stuff_check_ok` composable?